File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int m = mat.length;
        int sum = 0;
        for(int i=0;i<m;i++){
            if(i != m-i-1){
                sum += mat[i][m-i-1];
            }
            sum += mat[i][i];
        }
        return sum;
    }
}

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.c
int diagonalSum(int** mat, int matSize, int* matColSize) {
    if(matSize<2) return mat[0][0];
    int result=0,n=matSize;
    for(int i=0;i<matSize;i++){
            if(i == (n-i-1)){
                result = result+mat[i][n-i-1];
            }else{
                result = result+ mat[i][i]+ mat[i][n-i-1];
            }
    }
    return result;
}